utils/ixdtf/src/parsers/annotations.rs

```diff
@@ -54,18 +54,20 @@ pub(crate) fn parse_annotation_set<'a>(cursor: &mut Cursor<'a>) -> ParserResult<
 /// Parse any number of `KeyValueAnnotation`s
 pub(crate) fn parse_annotations<'a>(
     cursor: &mut Cursor<'a>,
-) -> ParserResult<(Option<&'a str>, Vec<Annotation<'a>>)> {
+) -> ParserResult<(Option<Annotation<'a>>, Vec<Annotation<'a>>)> {
     let mut annotations = Vec::default();
     let mut calendar = None;
     let mut calendar_crit = false;
 
     while cursor.check_or(false, is_annotation_open) {
         let kv = parse_kv_annotation(cursor)?;
 
+        // Check if the key is the registered key "u-ca".
         if kv.key == "u-ca" {
+            // If the calendar is not yet set, set calendar.
             if calendar.is_none() {
-                calendar = Some(kv.value);
                 calendar_crit = kv.critical;
+                calendar = Some(kv);
                 continue;
             }
```

Comment:
Suggestion: it would improve readability of this function to write it as

```rust
if kv.key == "u-ca" {
  match calendar {
    Some(seen_kv) if kv.critical || seen_kv.critical => {
      // Duplicate, critical calendar. This is an error.
      return Err(ParserError::CriticalDuplicateCalendar);
    }
    Some(_) => {
      // Duplicate, non-critical calendar. Add it to the annotations list.
    }
    None => {
      // First calendar annotation seen.
      calendar = Some(kv);
      continue;
    }
  }
}
annotations.push(kv);
```